polaris-react/src/components/Page/components/Header/Header.tsx

```diff
@@ -284,6 +305,7 @@ function PrimaryActionMarkup({
       shouldShowIconOnly(isNavigationCollapsed, primaryAction),
       {
         primary,
+        disabled: shouldOverrideDisableAction ?? primaryAction.disabled,
```

Comment:
Should this be an OR (i.e. `||`) rather than null coalescing? It seems `shouldOverrideDisableAction` would always be a boolean, so it should never fallback to `primaryAction.disabled`.